src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -220,23 +220,14 @@ internal async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool
                         }
 
                         // If the project has project references, we need to compute transitive origins for their packages
-                        List<LockFileTargetLibrary> projectReferences = targetsList.SelectMany(t => t.Libraries).Where(l => l.Type == LibraryType.Project).ToList();
-                        List<PackageReference> calculatedLibraryReferences = new List<PackageReference>(projectReferences.Count + calculatedInstalledPackages.Count);
-                        if (projectReferences.Count > 0)
-                        {
-                            foreach (var projectReference in projectReferences)
-                            {
-                                var packageReference = new PackageReference(
-                                    identity: new PackageIdentity(projectReference.Name, projectReference.Version),
-                                    targetFramework: new NuGetFramework(projectReference.Framework),
-                                    userInstalled: false,
-                                    developmentDependency: false,
-                                    requireReinstallation: false,
-                                    allowedVersions: new VersionRange(projectReference.Version));
-                                calculatedLibraryReferences.Add(packageReference);
-                            }
-                        }
-
+                        List<PackageReference> projectReferences = packageSpec
+                            .TargetFrameworks
+                            .SelectMany(f => GetProjectPackageReferences(f.FrameworkName, targetsList))
+                            .GroupBy(p => p.PackageIdentity)
+                            .Select(g => g.OrderBy(p => p.TargetFramework, FrameworkSorter).First())
```

Comment:
In here, you're basically deduplicating and choosing the first one. 

Can this cause issues when different frameworks of the same project have different references?
Would all of them be shown?

netstandard2.0
P1 -> P2 -> A, B
net472
P1 -> P2 -> A, C

This can be done in a follow up since it seems like this is not a new behavior, but how the old works as well.

Reply:
This is definitely going to cause issues with multiple frameworks, currently the PM UI uses this `FrameworkSorter` to decide which packages to show, this behavior is also present in top-level 😭 we need to design a better solution for this problem

Reply:
Here's my understanding. 

1. The PM UI will show *all* transitive packages. 
1. If a package is available with different versions in a different framework, the PM UI does not guarantee which  one of the versions is shown. 

My fear here was whether assumption #1 is correct.
It feels like a bigger issue.

Reply:
👀I'm not following the sort & choose first TFM logic. Can a test case be added that demonstrates why the first TFM works? In Nikolche's example I think this means only net472 will be added to `projectReferences` for group `p.PackageIdentity` ?

Reply:
> The PM UI will show all transitive packages.

You are right, it will show all top level and transitive's, but just as you mentioned in second point we do not guarantee which one is shown. There cannot be any duplicated package items in the same level (Top and transitive).

I thought the PM UI was choosing a framework before creating the list, at least we are displaying more information that I believed.

![image](https://github.com/user-attachments/assets/2d9525eb-2f20-47d9-acec-4d66706a9871)
![image](https://github.com/user-attachments/assets/e91c1ee6-46d5-40ae-bafa-c0d4a1e31d3c)

> If a package is available with different versions in a different framework, the PM UI does not guarantee which one of the versions is shown.

With the above configuration we see that the transitive origin data is using the other package version.

![image](https://github.com/user-attachments/assets/67524ac6-4be3-4246-bea2-bde705707837)